checkout-example-advanced/src/main/resources/static/adyenImplementation.js

```diff
@@ -0,0 +1,125 @@
+const clientKey = document.getElementById("clientKey").innerHTML;
+const type = document.getElementById("type").innerHTML;
+
+async function initCheckout() {
+  try {
+    const paymentMethodsResponse = await callServer("/api/getPaymentMethods");
+    const configuration = {
+      paymentMethodsResponse: filterUnimplemented(paymentMethodsResponse),
+      clientKey,
+      locale: "en_US",
+      environment: "test",
+      showPayButton: true,
+      paymentMethodsConfiguration: {
+        ideal: {
+          showImage: true,
+        },
+        card: {
+          hasHolderName: true,
+          holderNameRequired: true,
+          name: "Credit or debit card",
+          amount: {
+            value: 1000,
+            currency: "EUR",
+          },
+        },
+        paypal: {
+          amount: {
+            value: 1000,
+            currency: "USD",
+          },
+          environment: "test", // Change this to "live" when you're ready to accept live PayPal payments
+          countryCode: "US", // Only needed for test. This will be automatically retrieved when you are in production.
+          onCancel: (data, component) => {
+            component.setStatus('ready');
+          },
+        }
+      },
+      onSubmit: (state, component) => {
+        if (state.isValid) {
+          handleSubmission(state, component, "/api/initiatePayment");
+        }
+      },
+      onAdditionalDetails: (state, component) => {
+        handleSubmission(state, component, "/api/submitAdditionalDetails");
+      },
+    };
+    // `spring.jackson.default-property-inclusion=non_null` needs to set in
+    // src/main/resources/application.properties to avoid NPE here
+    const checkout = new AdyenCheckout(configuration);
+    checkout.create(type).mount(document.getElementById("payment"));
+  } catch (error) {
+    console.error(error);
+    alert("Error occurred. Look at console for details");
+  }
+}
+
+function filterUnimplemented(pm) {
```

Comment:
I would add a short comment to explain what this function does, and that must be updated if new payment methods must be enabled